src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java

```diff
@@ -329,13 +330,54 @@ private void addInvalidDisposalDateError(List<HearingTypeItem> hearingTypeItems,
             return;
         }
 
-        Optional<DateListedTypeItem> hearingTypeItem = findHearingTypeItem(hearingTypeItems, disposalDate);
-
-        if (hearingTypeItem.isEmpty() || !YES.equals(hearingTypeItem.get().getValue().getHearingCaseDisposed())) {
+        if (!disposedHearingWithSameDisposalDateOfJurisdictionExists(hearingTypeItems, disposalDate)) {
             errors.add(String.format(DISPOSAL_DATE_HEARING_DATE_MATCH, jurCode));
         }
     }
 
+    /**
+     * Checks if hearing type items has any hearing type which is disposed
+     * and has the same disposal date with the jurisdiction disposal date.
+     * @param hearingTypeItems All hearings
+     * @param disposalDate Disposal date of Jurisdiction
+     * @return boolean  if true then hearing list has disposed hearing with the
+     *                  same disposal date of jurisdiction.
+     */
+    private boolean disposedHearingWithSameDisposalDateOfJurisdictionExists(
+            List<HearingTypeItem> hearingTypeItems, String disposalDate) {
+        if (ObjectUtils.isNotEmpty(hearingTypeItems)) {
+            for (HearingTypeItem hearingTypeItem : hearingTypeItems) {
+                if (ObjectUtils.isNotEmpty(hearingTypeItem.getValue())
+                        && ObjectUtils.isNotEmpty(hearingTypeItem.getValue().getHearingDateCollection())
+                        && hearingDateCollectionHasDisposedHearingWithSameDisposalDateOfJurisdiction(
+                        hearingTypeItem.getValue().getHearingDateCollection(), disposalDate)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if any of the dateListedTypeItems has any hearing type which is disposed
+     * and has the same disposal date with the jurisdiction disposal date.
+     * @param dateListedTypeItems All dateListedType Items in a hearing
+     * @param disposalDate Disposal date of Jurisdiction
+     * @return boolean  if true then at least one of the listed type itemshas
+     *                  disposed hearing with the same disposal date of jurisdiction.
+     */
+    private boolean hearingDateCollectionHasDisposedHearingWithSameDisposalDateOfJurisdiction(
```

Comment:
method name is really long

Reply:
I use long method names to make them understandable to everyone. It is a polemic between developers, If this is not the practice that we generally use then I will make it shorter. https://softwareengineering.stackexchange.com/questions/409455/clean-code-long-names-instead-of-comments

Reply:
I am shortening the method names because in our projects we didn't use long method names. Thanks....